### ISSUE TITLE
Test Module for file i/o

### ISSUE THREAD

**Issue by peterplatten** [CST299TeamD/BearWatch#20]
Create a csv file, and zip some pictures into a zip file.

**Comment by peterplatten** [comment]
I'm going to attempt to emulate this using ripple & chrome - if it works it'll give a great headstart for when we can attempt it on the iPad.

**Comment by peterplatten** [comment]
Example code from: https://www.neontribe.co.uk/cordova-file-plugin-examples/

``` javascript
document.addEventListener('deviceready', onDeviceReady, false);  
function onDeviceReady() {  
    function writeToFile(fileName, data) {
        //data = JSON.stringify(data, null, '\t'); // not needed as I plan to pass it CSV's not JSON strings
        window.resolveLocalFileSystemURL(cordova.file.dataDirectory, function (directoryEntry) {
            directoryEntry.getFile(fileName, { create: true }, function (fileEntry) {
                fileEntry.createWriter(function (fileWriter) {
                    fileWriter.onwriteend = function (e) {
                        // pass a success callback here
                        console.log('Write of file "' + fileName + '"" completed.');
                    };

                    fileWriter.onerror = function (e) {
                        // you could hook this up with our global error handler, or pass in an error callback
                        console.log('Write failed: ' + e.toString());
                    };

                    var blob = new Blob([data], { type: 'text/plain' });
                    fileWriter.write(blob);
                }, errorHandler.bind(null, fileName));
            }, errorHandler.bind(null, fileName));
        }, errorHandler.bind(null, fileName));
    }

    //writeToFile('example.json', { foo: 'bar' });
    writeToFile('myfirst.csv', 'hello, world');
}
```

**Comment by peterplatten** [comment]
Update:
Application can write and read text files.

I was having an issue where I could create a csv file, however there were errors occurring when used as an attachment for emails. It took some diagnosis, but I've located the issue. I confirmed that the files were being created by using "adb shell" to find their location on the emulator. This location wasn't accessible by the email application. Switching the directory isn't an effective solution as we desire the files to be hidden from the user (no tampering), as well as accessible by other applications - an interesting line.

The solution:
There is an option for our application to send a data stream, and for the email application to build them as attachments when the user attempts to upload/submit/email the session. For the .csv files, this will be an improvement to performance, as we'll be instead having the email application write them to disk for us. For pictures, the camera plugin will create the picture files, then our application will need to re-import the picture files to our application and stream them to the email application, which will temporarily double the amount of space required for pictures on the iPad.

More testing is required, the Capstone Team wishes to close this issue to track the time spent during this sprint, and create a new issue for the added complexity for next sprint.
